src/pages/profiles/EditProfileForm.tsx

```diff
@@ -123,6 +123,23 @@ const EditProfileForm: FC<Props> = ({ profile }) => {
   });
 
   const getPayload = (values: EditProfileFormValues) => {
+    const supportedConfigKeys = getSupportedConfigKeys();
+    const additionalConfigKeys = Object.fromEntries(
+      Object.entries(profile.config).filter(
+        ([key]) => !supportedConfigKeys.has(key)
+      )
+    );
+
+    const supportedMainKeys = new Set([
+      "name",
+      "description",
+      "devices",
+      "config",
+    ]);
+    const additionalMainKeys = Object.fromEntries(
+      Object.entries(profile).filter(([key]) => !supportedMainKeys.has(key))
+    );
```

Comment:
This looks very similar to the one in `EditInstanceForm`. I was wondering if we could somehow reuse this code. Maybe one idea could be to have a `util` function that takes as input the `supportedMainKeys` (which are different between instances and profiles) and the `LxdInstance | LxdProfile` object, from which it can extract the `config` object. The function would then return the `additionalConfigKeys` and `additionalMainKeys` as a tuple, for example. What do you think?

Reply:
yeah, that is much better to put this logic into a reusable function.